docs/examples/advanced_optimizations.ipynb

```diff
@@ -0,0 +1,321 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "b96a9ef6",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Mean time: 27.82952880859375 ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Construct layer\n",
+    "basic_transformer = te.TransformerLayer(\n",
+    "    hidden_size,\n",
+    "    ffn_hidden_size,\n",
+    "    num_attention_heads,\n",
+    ")\n",
+    "basic_transformer.to(dtype=dtype).cuda()\n",
+    "\n",
+    "fp8_format = Format.HYBRID\n",
+    "fp8_recipe = DelayedScaling(\n",
+    "    fp8_format=fp8_format,\n",
+    "    amax_history_len=16,\n",
+    "    amax_compute_algo=\"max\",\n",
+    ")\n",
+    "# Training step\n",
+    "with te.fp8_autocast(enabled=True, fp8_recipe=fp8_recipe):\n",
+    "    y = basic_transformer(x, attention_mask=None)\n",
+    "y.backward(dy)\n",
+    "\n",
+    "# Measure step time\n",
+    "utils.speedometer(\n",
+    "    basic_transformer,\n",
+    "    x,\n",
+    "    dy,\n",
+    "    forward_kwargs = { \"attention_mask\": None },\n",
+    "    fp8_autocast_kwargs = { \"enabled\": True, \"fp8_recipe\": fp8_recipe },\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "11367f5b",
+   "metadata": {},
+   "source": [
+    "## Multi-GPU training\n",
+    "\n",
+    "<div class=\"alert alert-info\">\n",
+    "\n",
+    "<b>Summary</b>\n",
+    "    \n",
+    "We parallelize a Transformer layer with data, tensor, and sequence parallelism.\n",
+    "\n",
+    "</div>\n",
+    "\n",
+    "A variety of parallelism strategies can be used to enable multi-GPU training of Transformer models, often based on different approaches to distribute their $\\text{sequence_length} \\times \\text{batch_size} \\times \\text{hidden_size}$ activation tensors. The most common approach is data parallelism, which distributes along the $\\text{batch_size}$ dimension. By storing duplicate copies of the model on each GPU, the forward and backward passes of the training step can be done independently, followed by a gradient synchronization. A more advanced strategy is tensor parallelism, a type of model parallelism that distributes along the $\\text{hidden_size}$ dimension. This allows us to scale past the limits of data parallelism (typically $\\text{hidden_size} > \\text{batch_size}$) and to reduce the per-GPU memory usage (since model parameters are also distributed), but it also incurs the overhead of communicating activation tensors between GPUs at every step. For a more detailed explanation, please see the [Megatron-LM paper](https://arxiv.org/pdf/1909.08053.pdf). Finally, sequence parallelism distributes along the $\\text{sequence_length}$ dimension. Its communication pattern works synergistically with tensor parallelism and it can be used for operations that are not amenable to tensor parallelism (e.g. layer norm). For more details, please see [this paper](https://arxiv.org/pdf/2205.05198.pdf).\n",
```

Comment:
```suggestion
    "A variety of parallelism strategies can be used to enable multi-GPU training of Transformer models, often based on different approaches to distribute their $\\text{sequence_length} \\times \\text{batch_size} \\times \\text{hidden_size}$ activation tensors. The most common approach is data parallelism, which distributes along the $\\text{batch_size}$ dimension. By storing duplicate copies of the model on each GPU, the forward and backward passes of the training step can be done independently, followed by a gradient synchronization. A more advanced strategy is tensor parallelism, a type of model parallelism that distributes along the $\\text{hidden_size}$ dimension. This allows us to scale past the limits of data parallelism (typically $\\text{hidden_size} > \\text{batch_size}$) and to reduce the per-GPU memory usage (since model parameters are also distributed), but it also incurs the overhead of communicating activation tensors between GPUs at every step. For a more detailed explanation, please see the [Megatron-LM paper](https://arxiv.org/pdf/1909.08053.pdf). Finally, sequence parallelism distributes along the $\\text{sequence_length}$ dimension. This can be used when tensor parallelism is enabled in order to parallelize operations that run outside the tensor-parallel region (e.g. layer norm). For more details, please see [this paper](https://arxiv.org/pdf/2205.05198.pdf).\n",
```